django/views/i18n.py

```diff
@@ -318,3 +331,120 @@ def json_catalog(request, domain='djangojs', packages=None):
         'plural': plural,
     }
     return http.JsonResponse(data)
+
+
+class JavascriptCatalog(View):
+    """
+    Returns the selected language catalog as a javascript library.
+
+    Receives the list of packages to check for translations in the
+    kwargs `packages` key either from an infodict or as a +-delimited
+    string from the request. Default is 'django.conf'.
+
+    Additionally you can override the gettext domain for this view,
+    but usually you don't want to do that, as JavaScript messages
+    go to the djangojs domain. But this might be needed if you
+    deliver your JavaScript source from Django templates.
+    """
+    domain = 'djangojs'
+    packages = None
+
+    def get(self, request, *args, **kwargs):
+        locale = get_language()
+        domain = kwargs.get('domain', self.domain)
+        # If packages are not provided, default to all installed packages, as
+        # DjangoTranslation without localedirs is harvesting them all.
+        packages = kwargs.get('packages', '').split('+') or self.packages
+        paths = self.get_paths(packages) if packages else None
```

Comment:
Just to make sure, the previous behaviour would load the 'django.conf' package if packages is None? With this new change, you will load all installed apps and LOCALE_PATHS files, but 'django.conf' wouldn't be in it, so it may or may not be an issue, I can't really tell, since that package is used for the admin JS, so I don't think anyone would be relying on it for non admin pages. I actually like the change since in this way you don't have to manually specify the packages when all you want is your app's translations.